macros/sql/get_tables_by_pattern_sql.sql

```diff
@@ -16,13 +16,51 @@
 
 
 {% macro bigquery__get_tables_by_pattern_sql(schema_pattern, table_pattern, exclude='', database=target.database) %}
-    
-        select distinct
-            table_schema, table_name
 
-        from {{adapter.quote(database)}}.{{schema}}.INFORMATION_SCHEMA.TABLES
-        where table_schema = '{{schema_pattern}}'
-            and lower(table_name) like lower ('{{table_pattern}}')
-            and lower(table_name) not like lower ('{{exclude}}')
+    {% if '%' in schema_pattern %}
```

Comment:
This means that we actually introduced a slight regression by switching to using `like` operators instead of `=` operators

Reply:
Not sure I follow how this is a regression?

Question is, should we amend that line to be
```
{% if '%' in schema_pattern or '_' in schema_pattern %}
```
knowing we're going to get a _lot_ more false positives...

Reply:
> Not sure I follow how this is a regression?

Well I think previously for `get_relations_by_prefix` we used a strict `=` for the schema name.
```
where schema_name = 'a_c'
```
Now, we pass it to a `like` operator (i.e. [here](https://github.com/fishtown-analytics/dbt-utils/blob/6a9aa6f0d98696a2f5454ec9e0064796b21c8101/macros/sql/get_relations_by_prefix.sql#L5) and [here](https://github.com/fishtown-analytics/dbt-utils/blob/6a9aa6f0d98696a2f5454ec9e0064796b21c8101/macros/sql/get_tables_by_pattern_sql.sql#L11)
```
where schema_name like 'a_c'
```
This would match `a_c` and `abc`.

It's a teeny tiny regression for anyone that uses `get_relations_by_prefix`

Reply:
Ah, I see your point. To match _only_ `a_c`, the user should pass the pattern as `a\\_c`. I still think it's okay to ship this in a patch release, maybe link to the BQ doc in the readme.

Reply:
Oh this is across all warehouses! Not just BQ. I was just interrogating it here as part of this PR and it got me thinking.

I think let's just ship this as-is, and if someone notices the regression we can tell them how to fix it :)